### PR TITLE
Add optional parameters option for chat/completion/embedding.

### PR DESCRIPTION
**Pull request by JimmyAustin**
# Why

Some models have extra parameters that we want users to access. 

# What Changed
Added the ability to pass extra parameters to the option classes, that are then included in parameters. This caused some linting issues, which I've disabled for now, but if there's a better way of making these changes, I'm open to making them.

# Testing
Updated unit testing.

### CODE REVIEW

File: src/chat.ts [hunk 1]
```diff
@@ -35,6 +36,10 @@ export interface ChatOptions {
    * Defaults to 1024
    */
   maxOutputTokens?: number;
+
+  // Allows extra model specific parameters.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [key: string]: any;
```
Comment (masad-frost) [inline src/chat.ts:42]: I'd rather we move this to a sub field

`extraParams: Record<string, unknown>`

Comment (masad-frost) [reply src/chat.ts:42]: because otherwise it'll taint the whole interface

File: src/chat.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment*/
```
Comment (masad-frost) [inline src/chat.ts:1]: can we keep this line specific?

File: src/complete.ts [hunk 3]
```diff
@@ -35,6 +36,11 @@ export interface CompleteOptions {
    * Defaults to 1024
    */
   maxOutputTokens?: number;
+  9;
+
+  // Allows extra model specific parameters.
```
Comment (masad-frost) [inline src/complete.ts:41]: This should be in JSDoc comment format in order for the doc gen to pick it up

**Review by masad-frost — CHANGES_REQUESTED** [review]
a couple of nits

File: src/chat.ts (hunk 1 above)
Comment (JimmyAustin) [reply src/chat.ts:42]: These parameters are no different to the other ones, even at the API level. I don't think it makes sense to separate these out, but I'm also not sure what you mean by tainting the whole interface (not a TS expert). Can you elaborate?

Comment (masad-frost) [reply src/chat.ts:42]: I mean something like this https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgPIAczAPYgM7IDeAUMspALbrRxgCuUEA-AFzIh0UBG0A3KcgDaAawgBPNnjBRQAcwC6bOCDH8AvsWIJcU5NnRsMWHcgC8RAZXRRaDCGwDkABgB0AVgcAaZAHofyCmA8PGoAG1CIABNvEGxyMWpkPDgYCDAxSwALIP1jEEjsCDxYyAAPILA2AEZvP3ZsMGRGOHCxZGU9TBwQdvzkAHcIB0Y9YQG4Nv7gMEzkaeINLR1sCJdQ7FkACn0ASiA

We just lose a lot of the type safety and ergonomics.

> Some models have extra parameters that we want users to access.

We can also type-safe by using a union of options that are model specific.

Comment (JimmyAustin) [reply src/chat.ts:42]: We are ok with the lack of ability to catch typos. It'll throw an exception the moment they test it anyway. Lets go with the current approach.

File: src/chat.ts [hunk 4]
```diff
@@ -112,7 +119,7 @@ export async function chatStream(
   }
 
   return result.Ok(
-    pipe(res.value, async function* (source) {
```
Comment (JimmyAustin) [inline src/chat.ts:115]: Linter was complaining.

File: src/embed.ts [hunk 5]
```diff
@@ -17,6 +17,12 @@ const embeddingModels = ['textembedding-gecko'] as const;
 export interface EmbedOptions {
   model?: EmbedModel;
```
Comment (masad-frost) [inline src/embed.ts:18]: I'll fix this in a follow up, don't want this to be optional tbh

**Review by masad-frost — APPROVED** [review]
nice tests

File: src/embed.ts (hunk 5 above)
Comment (JimmyAustin) [reply src/embed.ts:18]: I can change it now.

**Pull request merged**